File: TD/TD2/Code/8.cpp
/**
 * Trouver une valeur choisie aléatoirement par le programme
 */
#include <cstdlib> // Contient la fonction rand()
#include <iostream>

int main() 
{
  int a;
  int b;
  int c;

  a = rand() % 100 + 1; // Utilisez % pour donner les limites (hack)
  b = 0;

  do {
    std::cout << "Donne moi une valeur : " << std::flush;
    std::cin >> c;

    if (c > a)
      std::cout << "C'est trop" << std::endl;
    else if (c < a)
      std::cout << "C'est pas assez" << std::endl;

    b++;
  } while (((c < a) || (c >a)) && (b < 6));

  if (c == a)
    std::cout << "Gagné" << std::endl;
  else
    std::cout << "Perdu" << std::endl;

  return 0;
}

File: TD/TD2/Code/Python/8.py
#!/usr/bin/python
# Trouver une valeur choisie aléatoirement par le programme
from random import randint

c = 0
i = 0
r = randint(1, 100)

while c != r and i < 6:
    print('Donne moi une valeur : ', end='')
    c = int(input())
    if c < r:
        print('C\'est pas assez !')
    elif c > r:
        print('C\'est trop !')
    i += 1

if c == r:
    print('Gagné !')
else:
    print('Perdu !')
